src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/TokenCredentialOptions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Core;
+using Azure.Identity;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    public class TokenCredentialOptions
+    {
+        private const string AuthToken = "AAD";
+        private const string AuthAuthorizationKey = "Authorization";
+        private const string AuthClientIdKey = "ClientId";
+
+        public TokenCredentialOptions(string authenticationString)
```

Comment:
nit: having a constructor which takes the connection string _and_ settable properties is not ideal (the property setters are never very useful). So either do one of two things:

1. Remove property setters
**OR**
2. Switch to a parameterless constructor and add a `public static TokenCredentialOptions Parse(string authenticationString)`
    - this way the auth string can be skipped entirely and a user directly constructs this options object (a bit more useful).